pgjdbc/src/main/java/org/postgresql/jdbc/BatchResultHandler.java

```diff
@@ -149,4 +150,52 @@ public void handleCompletion() throws SQLException {
   public ResultSet getGeneratedKeys() {
     return generatedKeys;
   }
+
+  private int[] uncompressUpdateCount() {
+    if (!(queries[0] instanceof BatchedQuery)) {
+      return updateCounts;
+    }
+    int totalRows = 0;
+    boolean hasRewrites = false;
+    for (Query query : queries) {
+      int batchSize = query.getBatchSize();
+      totalRows += batchSize;
+      hasRewrites |= batchSize > 1;
+    }
+    if (!hasRewrites) {
+      return updateCounts;
+    }
+
+    /* In this situation there is a batch that has been rewritten. Substitute
+     * the running total returned by the database with a status code to
+     * indicate successful completion for each row the driver client added
+     * to the batch.
+     */
+    int[] newUpdateCounts = new int[totalRows];
+    int offset = 0;
+    // In case of rewrite, we split into various queries
+    // It would be weird to have "no info" for some and 1 for another.
+    boolean isMultiValueBatchRewrite = queries[0].getBatchSize() > 1;
```

Comment:
If I remember correctly, it was not a technical issue but rather nonsense from a user point of view.
1 batch = 1 row. Fine.
2 batches = 2 no info.
3 batches = 2 no info, 1 row.
...
6 batches = 6 no info.
7 batches = 6 no info, 1 row.

Good luck with explaining users that behavior :)

But using hasRewrites is probably better than isMultiValueBatchRewrite.